server/app/auth/authorize_api_request.rb

```diff
@@ -0,0 +1,43 @@
+# app/auth/authorize_api_request.rb
+class AuthorizeApiRequest
+  def initialize(headers = {})
+    @headers = headers
+  end
+
+  # Service entry point - return valid user object
+  def call
+    {
+      user: user
+    }
+  end
+
+  private
+
+  attr_reader :headers
+
+  def user
+    # check if user is in the database
+    # memoize user object
+    @user ||= ApiUser.find(decoded_auth_token[:user_id]) if decoded_auth_token
+  # handle user not found
+  rescue ActiveRecord::RecordNotFound => e
+    # raise custom error
+    raise(
+      ExceptionHandler::InvalidToken,
+      ("#{Message.invalid_token} #{e.message}")
```

Comment:
Would this way of expressing it still be valid?

```suggestion
      "#{Message.invalid_token} #{e.message}"
```

Reply:
still valid, updated